presto-main/src/main/java/com/facebook/presto/sql/planner/plan/JoinNode.java

```diff
@@ -164,6 +176,18 @@ public PlanNode getRight()
                 .build();
     }
 
+    @JsonProperty("method")
+    public Optional<Method> getMethod()
+    {
+        return method;
+    }
+
+    public boolean isDistributed()
```

Comment:
Given this method is only used in one place and it is this one liner
`node.getMethod().orElseThrow(() -> new IllegalStateException("join method not yet set")) == DISTRIBUTED`, I'd prefer to see it inlined in the usage.